### DISCUSSION TITLE
What exactly is the tech stack of this project? Is it on React or Flask or something else?

### DISCUSSION THREAD

**Discussion by prajwalnh2006** [microsoft/sample-app-aoai-chatGPT#303] [Q&A]
It would be amazing to know the tech stack that this repo is based on. Which frameworks/technologies have been used in the code? Also can we please have some insights into the architecture of the code and what certain sections of the project folders do? Thank you so much in advance for all your help!

**Comment by sarah-widder** [comment]
Hi @prajwalnh2006 the frontend is using React. All of the frontend files are in the `frontend` folder. The backend is using Flask, the logic for this is in `app.py` in the root of the repo. For the backend interactions with CosmosDB to support the chat history feature, the logic is in the `backend` folder. Hope that helps :)

**Reply by lauramartinazure** [reply]
This might just be me groping around in the dark with React/Flask etc as I am primarily a data engineer, not an app developer, but when I edit anything in the frontend files - e.g. css files, changing which attributes of a citation to return etc - nothing happens in the deployed app despite the github actions build being successful. If I go directly into the javascript file in 'static' and make the same edits, these *do* come through. It took me an embarassingly long time to work that out! Any tips/directions for me here?

**Reply by pamelafox** [reply]
Did you re-build the code? See the note here: https://github.com/microsoft/sample-app-aoai-chatGPT?tab=readme-ov-file#deploy-with-the-azure-cli
